### ISSUE TITLE
Parquet support cleanup, consistency with DHE

### ISSUE THREAD

**Issue by niloc132** [deephaven/deephaven-core#2375]
This ticket covers assorted cleanup in the parquet implementation, mostly around code quality, as a pre-req for #2118. This will bring DHC in line with some DHE, which should make porting other features easier. The goal of this ticket is to track changes to make them consistent, other points listed here might be handled later.

Some fixes are obvious, so I'm just going to make them rather than list them here as I go over the diff, but others will be listed here for review:
 * ColumnWriterImpl.writePageV2 is unused
 * GenericArrayBuffer is unused
 * ParquetFIleWriter.addRowGroup is unused
 * RowGroupWriterImpl package-protected ctor is unused (blockWithPath too) except by addRowGroup above

Some followup we should do:
* Let ParquetInstructions offer guidance on precision and scale for BigDecimal columns
* Consider support for BigInteger, as decimals with scale 1(?), with metadata to indicate they are not actually decimals
* Look into DHE's ParquetTableWriter for grouping in tables, otherwise perhaps delete ParquetTableWriter.DestinationInfo

Other notes from Andy
* Check tests for bigdecimal vs serializable (that is, BigDecimal being encoded as a java serializable, rather than with the BigDecimal codec), ensure it always writes as bigdecimal - this might have been a bug introduced while working on adding BigInteger to begin with

**Comment by niloc132** [comment]
cc @jcferretti @rcaudy @abaranec

**Comment by rcaudy** [comment]
@abaranec I don't think we want to port changes to `ColumnChunkPageStore`. These were to accommodate compound regions (with binary search by offset to find rows) rather than the nested address-space division community uses, right?

**Comment by rcaudy** [comment]
> @abaranec I don't think we want to port changes to `ColumnChunkPageStore`. These were to accommodate compound regions (with binary search by offset to find rows) rather than the nested address-space division community uses, right?

Let the record show that Andy answered yes in Slack.
